src/app/views/authentication/Authentication.tsx

```diff
@@ -38,69 +44,112 @@ const Authentication = (props: any) => {
       const authResponse = await authenticationWrapper.logIn();
       if (authResponse) {
         setLoginInProgress(false);
-        dispatch(getAuthTokenSuccess(!!authResponse.accessToken))
-        dispatch(getConsentedScopesSuccess(authResponse.scopes))
+        dispatch(getAuthTokenSuccess(!!authResponse.accessToken));
+        dispatch(getConsentedScopesSuccess(authResponse.scopes));
       }
     } catch (error) {
       const { errorCode } = error;
-      dispatch(setQueryResponseStatus({
-        ok: false,
-        statusText: messages['Authentication failed'],
-        status: errorCode === 'popup_window_error'
-          ? translateMessage('popup blocked, allow pop-up windows in your browser')
-          : errorCode ? errorCode.replace('_', ' ') : '',
-        messageType: MessageBarType.error
-      }));
+      dispatch(
+        setQueryResponseStatus({
+          ok: false,
+          statusText: messages['Authentication failed'],
+          status: getErrorAndHint(errorCode),
+          messageType: MessageBarType.error,
+        })
+      );
       setLoginInProgress(false);
       telemetry.trackException(
         new Error(errorTypes.OPERATIONAL_ERROR),
         SeverityLevel.Error,
         {
           ComponentName: componentNames.AUTHENTICATION_ACTION,
-          Message: `Authentication failed: ${errorCode ? errorCode.replace('_', ' ') : ''}`,
-        });
+          Message: `Authentication failed: ${
+            errorCode ? errorCode.replace('_', ' ') : ''
```

Comment:
This will replace only the first occurrence of the `_` character. Try doing it like this

```suggestion
            errorCode ? errorCode.replace(/_/g, ' '): ''
```

Reply:
Yes. Let me fix that